solana/rpc/api.py

```diff
@@ -67,13 +401,1039 @@ def get_balance(self, pubkey: Union[PublicKey, str], commitment: Optional[Commit
+        ... )
+        >>> solana_client.simulate_transaction(tx_str)  # doctest: +SKIP
+        {'jsonrpc' :'2.0',
+         'result': {'context': {'slot': 218},
+         'value': {
+             'err': null,
+             'logs': ['BPF program 83astBRguLMdt2h5U1Tpdq5tjFoJ6noeGwaY3mDLVcri success']},
+         'id':1}
+        """  # noqa: E501 # pylint: disable=line-too-long
+        args = self._simulate_transaction_args(txn, sig_verify, commitment)
+        return self._provider.make_request(*args)
+
+    def set_log_filter(self, log_filter: str) -> types.RPCResponse:
+        """Sets the log filter on the validator.
+
+        :param log_filter: The new log filter to use.
+
+        >>> solana_client = Client("http://localhost:8899")
+        >>> solana_client.set_log_filter("solana_core=debug") # doctest: +SKIP
+        {'jsonrpc': '2.0', 'result': None, 'id': 1}
+        """
+        args = self._set_log_filter_args(log_filter)
+        return self._provider.make_request(*args)
+
+    def validator_exit(self) -> types.RPCResponse:
+        """Request to have the validator exit.
+
+        Validator must have booted with RPC exit enabled (`--enable-rpc-exit` parameter).
+
+        >>> solana_client = Client("http://localhost:8899")
+        >>> solana_client.validator_exit() # doctest: +SKIP
+        {'jsonrpc': '2.0', 'result': true, 'id': 1}
+        """
+        return self._provider.make_request(self._validator_exit)
+
+    def __post_send_with_confirm(self, resp: types.RPCResponse, conf_comm: Commitment) -> types.RPCResponse:
+        resp = self._post_send(resp, self._provider)
+        self._provider.logger.info(
+            "Transaction sent to %s. Signature %s: ", self._provider.endpoint_uri, resp["result"]
+        )
+        return self.__confirm_transaction(resp["result"], conf_comm)
+
+    def __confirm_transaction(self, tx_sig: str, commitment: Optional[Commitment] = None) -> types.RPCResponse:
+        # TODO: Use websockets and check confirmation with onSignature subscription.
+        TIMEOUT = 30  # 30 seconds  pylint: disable=invalid-name
+        elapsed_time = 0
+        while elapsed_time < TIMEOUT:
+            sleep_time = 3
+            if not elapsed_time:
+                sleep_time = 7 if (commitment or self._commitment) == Finalized else sleep_time
+                sleep(sleep_time)
+            else:
+                sleep(sleep_time)
+
+            resp = self.get_confirmed_transaction(tx_sig)
+            if resp["result"]:
+                break
+            elapsed_time += sleep_time
+
+        if not resp["result"]:
+            raise Exception("Unable to confirm transaction %s" % tx_sig)
+        err = resp.get("error") or resp["result"].get("meta").get("err")
+        if err:
+            self._provider.logger.error("Transaction error: %s", err)
+
+        return resp
+
+
+class AsyncClient(_ClientCore):  # pylint: disable=too-many-public-methods
```

Comment:
Would it be possible to split into three different files?

- solana/rpc/core.py
- solana/rpc/api.cpy (old synchornous client)
- solana/rpc/async_api.py (new async client)

Reply:
Will do

Reply:
In the same spirit, I'll give AsyncToken its own file too